client/src/utils/auth.js

```diff
@@ -4,4 +4,68 @@
 // Importing the necessary packages
 // ================================================================
 import { jwtDecode } from 'jwt-decode';
-// ================================================================
\ No newline at end of file
+// ================================================================
+
+const Auth = {
+    // Function to set token to localStorage
+    // ================================================================
+    login: (token) => {
+        localStorage.setItem('id_token', token); // Store the JWT token in local storage under the key 'id_token' 
+        window.location.assign('/'); // Redirect user to main page
+    },
+    // ================================================================
+
+    // Function to get the token from localStorage
+    // ================================================================
+    getToken: () => {
+        return localStorage.getItem('id_token');
+    },
+    // ================================================================
+
+    // Function to remove the token from localStorage
+    // ================================================================
+    logout: () => {
+        localStorage.removeItem('id_token');
+        window.location.reload();
+    },
+    // ================================================================
+
+    // Function to get the user data from the token
+    // ================================================================
+    getUserData: () => {
+        const token = Auth.getToken();
+
+        if (token) {
+            return jwtDecode(token).data; // If a token exists, decode it to extract user data
+        }
+        return null;
+    },
+    // ================================================================
+
+    // Function to check if the user is logged in
+    // ================================================================
+    isLoggedIn: () => {
```

Comment:
please replace it with the following:
loggedIn() {
        const token = this.getToken();
        return token && !this.isTokenExpired(token) ? true : false;
    }